src/channel.cr

```diff
@@ -653,4 +653,65 @@ class Channel(T)
       raise ClosedError.new
     end
   end
+
+  class TimeoutAction
```

Comment:
Can be a struct, because immutable

Reply:
Since others `SelectAction` are classes it is better to keep them all like that so the representation is a single pointer instead of a mixed union of value and reference types.